server/src/main/java/org/elasticsearch/index/engine/MergeTracking.java

```diff
@@ -41,6 +41,9 @@ public class MergeTracking {
     private final Set<OnGoingMerge> onGoingMerges = ConcurrentCollections.newConcurrentSet();
     private final Set<OnGoingMerge> readOnlyOnGoingMerges = Collections.unmodifiableSet(onGoingMerges);
 
+    private final Set<OnGoingMerge> queuedMerges = ConcurrentCollections.newConcurrentSet();
+    private final Set<OnGoingMerge> readOnlyQueuedMerges = Collections.unmodifiableSet(queuedMerges);
```

Comment:
That's an interesting idea! I guess we can piggy back on the fact that `UnmodifiableCollection` is just a shell and the changes to `queuedMerges` are guaranteed to be visible via the `readOnlyQueuedMerges` reference.